apps/backend/src/factory/xlsx/stfc/StfcFapDataRow.ts

```diff
@@ -1,20 +1,46 @@
+import { stripHtml } from 'string-strip-html';
+
+import { StfcUserDataSource } from '../../../datasources/stfc/StfcUserDataSource';
 import { FapProposal } from '../../../models/Fap';
 import { FapMeetingDecision } from '../../../models/FapMeetingDecision';
 import { InstrumentWithAvailabilityTime } from '../../../models/Instrument';
 import { Proposal } from '../../../models/Proposal';
+import { QuestionaryStep } from '../../../models/Questionary';
+import { Review } from '../../../models/Review';
 import { TechnicalReview } from '../../../models/TechnicalReview';
 import { RowObj } from '../fap';
 import { getDataRow } from '../FapDataRow';
 
-export function getStfcDataRow(
+const stfcUserDataSource = new StfcUserDataSource();
+
+export async function getStfcDataRow(
   piName: string,
   proposalAverageScore: number,
   instrument: InstrumentWithAvailabilityTime,
   fapMeetingDecision: FapMeetingDecision | null,
   proposal: Proposal | null,
   technicalReview: TechnicalReview | null,
-  fapProposal?: FapProposal
+  fapProposal: FapProposal | null,
+  proposalAnswers: QuestionaryStep[] | null,
+  reviews: Review[] | null
 ) {
+  const daysRequested = proposalAnswers
+    ?.find((step) =>
+      step.fields.find(
+        (answer) => answer.question.naturalKey === 'days_requested'
+      )
+    )
+    ?.fields.find((answer) => answer.question.naturalKey === 'days_requested')
+    ?.value.value;
```

Comment:
This feels like there's a bit of duplicate filtering going on to find the questionary with the days_requested answer in it, and then the days_requested answer. I think it could simplify to something like:
```
proposalAnswers
  .flatMap(step => step.fields)
  .find(answer => answer.question.naturalKey === 'days_requested')
```

It also seems odd that only certain steps in the call chain are null conditional. I'd have thought if we need to have the first value protected, then the second one would also have a possibility of being called on null, and would need the same protection.

Reply:
When I was writing this I thought the double `find` is slightly more efficient than the `flatMap` as I does not need to construct a new array? This way is much more readable and I guess the cost différance cant be that much.